src/main/java/be/seeseemelk/mockbukkit/entity/ParrotMock.java

```diff
@@ -0,0 +1,105 @@
+package be.seeseemelk.mockbukkit.entity;
+
+import be.seeseemelk.mockbukkit.ServerMock;
+import com.google.common.base.Preconditions;
+import org.bukkit.entity.AnimalTamer;
+import org.bukkit.entity.EntityType;
+import org.bukkit.entity.Parrot;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.UUID;
+
+public class ParrotMock extends AnimalsMock implements Parrot
+{
+
+	private @NotNull Parrot.Variant variant = Variant.RED;
+	private boolean isDancing = false;
+	private boolean isSitting = false;
+	private boolean isTamed = false;
+	private AnimalTamer animalTamer = null;
+
+	/**
+	 * Constructs a new {@link ParrotMock} on the provided {@link ServerMock} with a specified {@link UUID}.
+	 *
+	 * @param server The server to create the entity on.
+	 * @param uuid   The UUID of the entity.
+	 */
+	public ParrotMock(@NotNull ServerMock server, @NotNull UUID uuid)
+	{
+		super(server, uuid);
+	}
+
+	@Override
+	public @NotNull Variant getVariant()
+	{
+		return this.variant;
+	}
+
+	@Override
+	public void setVariant(@NotNull Variant variant)
+	{
+		Preconditions.checkNotNull(variant, "Variant cannot be null");
+		this.variant = variant;
+	}
+
+	@Override
+	public boolean isDancing()
+	{
+		return this.isDancing;
+	}
+
+	public void setDancing(boolean isDancing)
+	{
+		this.isDancing = isDancing;
+	}
```

Comment:
`#isDancing` only returns true when there's a jukebox playing a record nearby. I don't believe we have jukeboxes implemented yet, so this shouldn't be implemented until we do.

Reply:
The juckbox is well implemented, but there are 3 methods that aren't "done", including `startPlaying`, and I wonder if it isn't this method that sets the peroquer's dancing to `true` (I haven't checked on papermc yet).

Reply:
Should I add TO DO comment in setDancing method and make it throw new UnimplementedOperationException()?

Reply:
setDancing should be removed, and isDancing should throw, along with a comment about why it isn't implemented.